src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/QualityGuidelines/AssigningSymbolAndItsMemberInSameStatementTests.cs

```diff
@@ -47,17 +73,40 @@ public class Test
     public void Method()
     {
         C a = new C(), b = new C(), c;
-        a.Property = c = a = b;
+        [|a.Property|] = c = a = b;
```

Comment:
@mavasani What should be the codefix(es) for this?

Reply:
> // For something like `a.x = a = b;`, we offer 3 code fixes:
            // First:
            //     a = b;
            //     a.x = b;
            // Second:
            //     a = b;
            //     a.x = a;
            // Third: (Not currently implemented)
            //     var temp = a;
            //     a = b;
            //     temp.x = b;

So I would say that you should end-up with:

```csharp
a = b;
c = a;
a.Property = b;
```

and/or

```csharp
a = b;
c = a;
a.Property = c;
```

Basically we take the far left or far right member.